File: bmm.py
# a: الگوریتم محاسبه ب.م.م : بزرگترین مقسوم علیه ( شمارنده ) مشترک 

# (num1 , num2 ) 
num1:int = int( input("Enter number 1 of (num1 , num2): ") )
num2:int = int( input("Enter number 2 of (num1 , num2): ") )

counter = max(num1 , num2)  # iter in (num1 , num2 )

result = None

while ( counter > 1):
	if ( (num1 % counter) == 0 ) and ( (num2 % counter) == 0 ):
		if result == None:
			result = counter			
	
	counter -= 1
	
if result == None:
	print("The Numbr <1>")
else:
	print( f"The Number is <{result}>")

File: bmm.c
// see the bmm.py file

#include <stdio.h>

#define max(x, y) (((x) > (y)) ? (x) : (y))

int main(){

int num1;
int num2;

printf("Enter The number 1 and number 1 ( (number 1 , number 2 )): ");
scanf("%d %d" , &num1 , &num2);

int result = 0;

for ( int counter = max(num1 , num2); counter >= 1; counter--){
	if ( ((num1 % counter) == 0 ) && ((num2 % counter) == 0 ) ){
		if ( result == 0)
			result = counter;
	}
}
 
if (result == 0)
 	printf("The b.m.m is number <1>");
else
 	printf("The b.m.m is number <%d>", result);
 	
 return 0;
 }
